arch/x86/entry.S

```diff
@@ -154,6 +166,16 @@ ENTRY(enter_usermode)
     IRET
 END_FUNC(enter_usermode)
 
+ENTRY(syscall_exit)
+    POPF
+
+    /* Save exit code to return value register (AX) */
+    mov %_ASM_SI, cpu_regs_ax(%_ASM_SP)
+    RESTORE_ALL_REGS
+
+    ret
```

Comment:
Related to my confusion about `terminate_user_task`, where does this `RET` lead to? It's clearly intended to end up in kernel code, as we're running with kernel GS and CR3. But I fail to see where, clearly lacking some context here.

Reply:
The RET returns to the caller of `enter_usermode()`, see the previous comment.